roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,354 @@
+            except OpenShiftCheckException as exc:
+                self.register_failure(exc)
+
+        subnets = {hostsubnet['metadata']['name']: hostsubnet['subnet']
+                   for hostsubnet in self.get_resource('hostsubnets')}
+
+        subnet = subnets.get(node_name, None)
+        if subnet is None:
+            self.register_failure('Node %s has no hostsubnet.' % node_name)
+            return
+        subnet = six.text_type(subnet)
+        address = ipaddress.ip_network(subnet)[1]
+
+        for remote_node in self.get_resource('nodes'):
+            remote_node_name = remote_node['metadata']['name']
+            if remote_node_name == node_name:
+                continue
+
+            remote_subnet = subnets.get(remote_node_name, None)
+            if remote_subnet is None:
+                continue
+            remote_subnet = six.text_type(remote_subnet)
+            remote_address = ipaddress.ip_network(remote_subnet)[1]
+
+            self.save_command_output(
+                'trace_node_%s_to_node_%s' % (node_name, remote_node_name),
+                ['/bin/docker', 'exec', ovs_cid,
+                 '/bin/ovs-appctl', 'ofproto/trace', 'br0',
+                 'in_port=2,reg0=0,ip,nw_src=%s,nw_dst=%s' %
+                 (address, remote_address)])
+
+            try:
+                self.save_command_output('ping_node_%s_to_node_%s' %
+                                         (node_name, remote_node_name),
+                                         ['/bin/ping', '-c', '1', '-W', '2',
+                                          str(remote_address)])
+            except OpenShiftCheckException as exc:
+                self.register_failure('Node %s cannot ping node %s.' %
+                                      (node_name, remote_node_name))
+
+    def save_service_logs(self, service_name):
+        """Save the first 5 minutes of logs after the specified service started
+        and the last 5 minutes of logs for that service."""
+        out = self.read_command_output(['systemctl', 'show', service_name,
+                                        '-p', 'ExecMainStartTimestamp'])
+        start_timestamp = out.strip().split('=', 1)[1]
+        if len(start_timestamp) == 0:
+            self.register_failure('%s is not started.' % service_name)
+            return
+
+        start_time = datetime.datetime.strptime(start_timestamp,
+                                                '%a %Y-%m-%d %H:%M:%S %Z')
+        time_fmt = '%Y-%m-%d %H:%M:%S'
+        since = start_time.strftime(time_fmt)
+        until = (start_time + datetime.timedelta(minutes=5)).strftime(time_fmt)
+        start_logs = self.read_command_output(['/bin/journalctl',
+                                               '"--since=%s"' % since,
+                                               '"--until=%s"' % until])
+        last_logs = self.read_command_output(['/bin/journalctl', '-u', service_name,
+                                              '--since=-5m'])
```

Comment:
As mentioned before, get last 5 minutes of logs from last log time or get fixed number of lines from the last for it to be useful.

Reply:
Fixed to get the last 5 minutes of logs.